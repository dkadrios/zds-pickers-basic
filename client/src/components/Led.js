import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styles from '../styles/midiActivity.scss'

const Led = ({ label, lit, isMidiActivity, layout }) => {
  const className = classNames({
    [styles.led]: !lit && !isMidiActivity,
    [styles.ledBlink]: lit && !isMidiActivity,
    [styles.ledPower]: !lit && isMidiActivity,
    [styles.ledPowerBlink]: lit && isMidiActivity,
  })
  return (
    <figure
      className={styles.midiActivity}
      style={{ justifyContent: layout === 'left' ? 'flex-start' : 'flex-end' }}
    >
      {layout === 'right' && <header>{label}</header>}
      <div className={className} />
      {layout === 'left' && <header>{label}</header>}
    </figure>
  )
}

Led.propTypes = {
  label: PropTypes.string.isRequired,
  lit: PropTypes.bool.isRequired,
  isMidiActivity: PropTypes.bool,
  layout: PropTypes.oneOf(['left', 'right']),
}
Led.defaultProps = {
  isMidiActivity: false,
  layout: 'left',
}

export default Led
