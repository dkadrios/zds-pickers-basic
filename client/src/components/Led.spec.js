import Led from './Led'

describe('Led tests', () => {
  it('renders correctly', () => {
    shallowExpect(Led)({ label: 'label', lit: false }).toMatchSnapshot()
  })

  it('should render when lit', () => {
    shallowExpect(Led)({ label: 'label', lit: true }).toMatchSnapshot()
  })
})
