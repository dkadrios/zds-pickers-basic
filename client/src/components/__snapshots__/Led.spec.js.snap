// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Led tests renders correctly 1`] = `
<figure
  style={
    Object {
      "justifyContent": "flex-start",
    }
  }
>
  <div
    className=""
  />
  <header>
    label
  </header>
</figure>
`;

exports[`Led tests should render when lit 1`] = `
<figure
  style={
    Object {
      "justifyContent": "flex-start",
    }
  }
>
  <div
    className=""
  />
  <header>
    label
  </header>
</figure>
`;
